src/utils/tests/getElementByDataTestIdDeep.jest.ts

```diff
@@ -0,0 +1,16 @@
+import { createDataTestIdSelector, walkShadowDom } from './shadowDomTraversal';
+
+/**
+ * Recursively searches the document and all shadow roots for the first element
+ * matching the given data-testid using Jest/JSDOM.
+ *
+ * @param {string} dataTestId - The value of the data-testid attribute to search for.
+ * @param {Document|ShadowRoot} root - The root to search from (defaults to document).
+ * @returns {Element|null} - The found element, or null.
+ */
+export function getElementByDataTestIdDeepJest(
```

Comment:
Can't we use an arrow function here? Also, maybe we could have an object as argument for this helper, instead of multiple arguments. Would be more future proof.

Reply:
To avoid name collision on exports and imports